specification/data-faas.md

```diff
@@ -0,0 +1,83 @@
+# Semantic conventions for FaaS spans
+
+This document defines how to describe an instance of a function that runs without provisioning or managing of servers (also known as serverless) with spans.
+
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [General Attributes](#general-attributes)
+  * [Difference between execution and instance](#difference-between-execution-and-instance)
+- [Function Type](#function-type)
+  * [Datasource](#datasource)
+  * [HTTP](#http)
+  * [PubSub](#pubsub)
+  * [Timer](#timer)
+  * [Other](#other)
+
+<!-- tocstop -->
+
+# General Attributes
+
+Span `name` should be set to the function name being executed. Depending on the value of the `faas.trigger` attribute, additional attributes MUST be set. For example, an `http` trigger MUST follow the [HTTP Server semantic conventions](data-http.md#http-server-semantic-conventions). For more information, refer to the [Function Type](#function-type) section.
+
+If Spans following this convention are produced, a Resource of type `faas` MUST exist following the [Resource semantic convention](data-resource-semantic-conventions.md#function-as-a-service). 
+
+| Attribute name  | Notes  and examples  | Required? |
+|---|---|--|
+| `component` | Denotes the type of the span and needs to be `"faas"` | Yes | 
+| `faas.trigger` | Type of the trigger on which the function is executed. <br > it SHOULD be one of the following strings: "datasource", "http", "pubsub", "timer", or "other". | Yes |
+| `faas.execution` | String containing the unique execution id of the function. E.g. `af9d5aa4-a685-4c5f-a22b-444f80b3cc28` | No |
+
+## Difference between execution and instance
+
+For performance reasons (e.g. [AWS lambda], or [Azure functions]), FaaS providers allocate an execution environment for a single instance of a function that is used to serve multiple requests.
+Developers exploit this fact to solve the **cold start** issue, caching expensive resource computations between different function executions. 
+Furthermore, FaaS providers encourage this behavior, e.g. [Google functions].
+Therefore, the span attribute `faas.execution` differs from the resource attribute `faas.instance` in the following:
+
+- `faas.execution` refers to the current request ID handled by the function;
+- `faas.instance` refers to the execution environment ID of the function.
+
+[AWS lambda]: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
+[Azure functions]: https://docs.microsoft.com/en-us/azure/azure-functions/manage-connections#static-clients
+[Google functions]: https://cloud.google.com/functions/docs/concepts/exec#function_scope_versus_global_scope
+
+# Function Type
+
+This section describes how to handle the span creation and additional attributes based on the value of the attribute `faas.trigger`.
+
+## Datasource
+The function is meant to answer to a single event generated by a datasource, e.g. DynamoDB, Cloud Storage, Cosmos DB.
```

Comment:
Is the lambda function really serving a database request in this case? I have never used this so I'm not sure.

Reply:
You can configure lambda to be triggered when a row is inserted into a DB, sort of like a stored procedure. It does not serve the request, just may handle some business logic like updating a cache or sending an email or something.

example use case: If a row is inserted into the `PasswordResetTokens` table, send the user an email with a password reset link.